client/termsetting/handlers/geneExpression.ts

```diff
@@ -26,13 +26,17 @@ export async function getHandler(self) {
 }
 
 export async function fillTW(tw: GeneExpressionTW, vocabApi: VocabApi, defaultQ: NumericQ | null = null) {
-	if (typeof tw.term.gene != 'string' || !tw.term.gene) throw 'geneExpression tw.term.gene must be non-empty string'
+	if (typeof tw.term !== 'object') throw 'tw.term is not an object'
+	if (!tw.term.gene && !tw.term.name) throw 'no gene or name present'
+	if (!tw.term.gene) tw.term.gene = tw.term.name
+	if (!tw.term.gene || typeof tw.term.gene != 'string') throw 'geneExpression tw.term.gene must be non-empty string'
+
 	if (!tw.term.name) tw.term.name = tw.term.gene // auto fill if .name is missing
 
 	if (!tw.q?.mode) tw.q = { mode: 'continuous' } // supply default q if missing
 	if (defaultQ) copyMerge(tw.q, defaultQ) // override if default is given
 
-	if (!tw.term.bins) {
+	if (tw.q.mode !== 'continuous' && !tw.term.bins) {
```

Comment:
should revert this. a missing bin must be filled in fillTW. doing it in numeric.toggle.ts is outside of notify cycle, resulting in term.bins{} only remains in termsetting instance and not made available to plot data, thus the barchart err

Reply:
> should revert this. a missing bin must be filled in fillTW. doing it in numeric.toggle.ts is outside of notify cycle, resulting in term.bins{} only remains in termsetting instance and not made available to plot data, thus the barchart err @congyu-lu @siosonel
> 
> may implement #1774 to mitigate the lag issue

Should be fixed in https://github.com/stjude/proteinpaint/pull/1870, but not having to revert. Can discuss the fix or other issues there. The main issue was `term.bins` was being set on a copy of `tw` instead of the original data when using `controls.term1.js`, that's why the changes were not being propagated.